src/libs/actions/TaxRate.ts

```diff
@@ -0,0 +1,129 @@
+import Onyx from 'react-native-onyx';
+import * as API from '@libs/API';
+import type {CreatePolicyTaxParams} from '@libs/API/parameters';
+import {WRITE_COMMANDS} from '@libs/API/types';
+import CONST from '@src/CONST';
+import * as ErrorUtils from '@src/libs/ErrorUtils';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {TaxRate, TaxRates} from '@src/types/onyx';
+import type {PendingAction} from '@src/types/onyx/OnyxCommon';
+import type {OnyxData} from '@src/types/onyx/Request';
+
+/**
+ * Get tax value with percentage
+ */
+function getTaxValueWithPercentage(value: string): string {
+    return `${value}%`;
+}
+
+function covertTaxNameToID(name: string) {
+    return `id_${name.toUpperCase().replaceAll(' ', '_')}`;
+}
+
+/**
+ * Get new tax ID
+ */
+function getNextTaxCode(name: string, taxRates?: TaxRates): string {
+    const newID = covertTaxNameToID(name);
+    if (!taxRates?.[newID]) {
+        return newID;
+    }
+
+    // If the tax ID already exists, we need to find a unique ID
+    let nextID = 1;
+    while (taxRates?.[covertTaxNameToID(`${name}_${nextID}`)]) {
+        nextID++;
+    }
+    return covertTaxNameToID(`${name}_${nextID}`);
+}
+
+function createPolicyTax(policyID: string, taxRate: TaxRate) {
```

Comment:
We kept the create policy tags / categories in the policy.ts actions file. I think we should keep this in as well @luacmartins

Reply:
Ok thanks for the link. I think we should then follow up with similar update for other Policy features. @luacmartins

Reply:
Yea, I think we can clean this up later and create an utils file for each of them. Created issue [here](https://github.com/Expensify/App/issues/38477)